src/explorer/helpers/node-icon.ts

```diff
@@ -1,12 +1,30 @@
 import { NodeType } from "~app/types"
 import fileImage from "~images/file.png"
+import textFileImage from "~images/text_file.png"
+import audioFileImage from "~images/audio_file.png"
 import folderImage from "~images/folder.png"
+import fullFolderImage from "~images/full_folder.png"
 import unknownImage from "~images/unknown.png"
 
-export const nodeIcon = (type: NodeType) => {
+export const nodeIcon = (type: NodeType, name: string) => {
+  if (type === NodeType.FOLDER || type === NodeType.FULL_FOLDER) {
```

Comment:
instead of adding `FULL_FOLDER` as a new type, we should let `nodeIcon` do the legwork and find out if a folder `hasChildren` or not. This means that when we search for nodes within a folder, we also need to lookup how many nodes are withing the child nodes 😛. This feature is going to keep us busy for a bit, as we would want to optimize it a bit.